File: problems/p207.py
from time import perf_counter
start = perf_counter()

def is_power_of_two(n):
    return n != 0 and n & (n - 1) == 0;

c, x = 1, 3
while c/x >= 1/12345:
    if is_power_of_two(x):
        c += 1
    x += 1

x += 1
print('Solution:', x * (x - 1))

end = perf_counter()
print(end - start, 'seconds to run')

File: problems/201-300/p207.c
#include <stdio.h>
#include <time.h>

int main(int argc, char *argv[]) {
    clock_t start = clock();

    int c = 1, x = 3;
    while ((double) c/x >= 1.0 / 12345.0) {
        if ((x & (x - 1)) == 0) {
            c++;
        }
        x++;
    }
    printf("Solution: %lu\n", (long) x * (x + 1));

    clock_t end = clock();
    double time = (double) (end - start) / CLOCKS_PER_SEC;
    printf("%f seconds to run\n", time);
}
